flow/flow.cpp

```diff
@@ -127,10 +129,27 @@ enum class FileOutputMode {
 
 
 
+void ensureOutputDirExists(const std::string& cmdline_output_dir)
+{
+    // ensure that the output directory exists and that it is a directory
```

Comment:
This is a prime example of the type of comments which in my opinion just reduces the quality of the code. To be on the safe side I hardly ever write comments; you could claim that is the other ditch :-)

Reply:
The code is copied straight from the vaguard class, where it was part of a longer function, I agree it is redundant here, as it was put into an appropriately-named little function.